pages/admin/reports/components/TotalEarnings.tsx

```diff
@@ -0,0 +1,260 @@
+                shadow='sm'
+            >
+                <Text textAlign='center' fontSize='2xl' m='6' fontWeight='bold'>
+                    Ganancias por tratamiento
+                </Text>
+
+                {filteredStats.length === 0 ? (
+                    <Text textAlign='center' color='gray.500' py={10}>
+                        {
+                            // si falta cualquiera de las dos fechas, muestro siempre “No hay datos…”
+                            !dateRange.startDate || !dateRange.endDate
+                                ? 'No hay datos para mostrar en el período seleccionado'
+                                : appointments.length === 0
+                                ? 'Cargando datos...'
+                                : 'No hay datos para mostrar en el período seleccionado'
+                        }
+                    </Text>
+                ) : (
+                    <Box
+                        height={isMobile ? '400px' : '500px'}
+                        width='100%'
+                        px={0}
+                    >
+                        <ResponsiveContainer width='100%' height='100%'>
+                            <BarChart
+                                data={filteredStats}
+                                layout='horizontal'
+                                margin={{
+                                    top: 20,
+                                    right: isMobile ? 5 : 10,
+                                    left: isMobile ? 5 : 10,
+                                    bottom: isMobile ? 50 : 60,
+                                }}
+                                barGap={isMobile ? 1 : 4}
+                            >
+                                <XAxis
+                                    dataKey='name'
+                                    height={isMobile ? 50 : 80}
+                                    tickFormatter={(value) =>
+                                        formatTreatmentName(value) || ''
+                                    }
+                                    tick={{ fontSize: isMobile ? 10 : 12 }}
+                                    angle={isMobile ? -35 : 0}
+                                />
+                                <YAxis
+                                    yAxisId='left'
+                                    allowDecimals={false}
+                                    tickFormatter={(value) =>
+                                        Math.floor(value).toString()
+                                    }
+                                />
+                                <YAxis
+                                    yAxisId='right'
+                                    orientation='right'
+                                    tickFormatter={(value) =>
+                                        `$${Math.floor(value).toLocaleString()}`
+                                    }
+                                />
+                                <Tooltip
```

Comment:
Extract the formatting to a different function or util
```suggestion
    const formatTooltipValue = (value: number, name: string) => {
    return name === 'Realizaciones'
        ? `${Math.floor(value)}`
        : `$${Math.floor(value).toLocaleString()}`;
};

formatter={(value, name) => [formatTooltipValue(Number(value), name), name]}

```

Reply:
Using the chart examples provided by recharts, I realized that it has an intelligent default behavior: the <Tooltip /> automatically detects the name and dataKey props of each <Bar /> to display the values ​​without the need for a formatter, so in this case it is not necessary to apply it.